lib/tests/misc.nix

```diff
@@ -4432,4 +4432,82 @@ runTests {
     expected = "/non-existent/this/does/not/exist/for/real/please-dont-mess-with-your-local-fs/default.nix";
   };
 
+  # Tests for cross index utilities
+
+  testRenameCrossIndexFrom = {
+    expr = lib.renameCrossIndexFrom "pkgs" {
+      pkgsBuildBuild = "buildBuild";
+      pkgsBuildHost = "buildHost";
+      pkgsBuildTarget = "buildTarget";
+      pkgsHostHost = "hostHost";
+      pkgsHostTarget = "hostTarget";
+      pkgsTargetTarget = "targetTarget";
+    };
+    expected = {
+      buildBuild = "buildBuild";
+      buildHost = "buildHost";
+      buildTarget = "buildTarget";
+      hostHost = "hostHost";
+      hostTarget = "hostTarget";
+      targetTarget = "targetTarget";
```

Comment:
To ensure the test catches logic errors I think we want the expected names and values here to not match each other.